src/blr/cli/extractbarcode.py

```diff
@@ -3,45 +3,110 @@
 separated by an underline.
 """
 import logging
+import sys
+import dnaio
 
 import blr.utils as BLR
 
 logger = logging.getLogger(__name__)
 
 
+def input_files(inputs, interleaved):
+    """
+    Checks input arguments to see if they are sound and assigns them to input1 and input2
+    :param inputs: list of strings. Input files or paths given.
+    :param interleaved: boolean. If input is interleaved or not.
+    :return:
+    """
+    input1, input2 = (None, None)
+    if len(inputs) == 0:
+        sys.exit(f'EXITING: No inputs given!')
+    elif len(inputs) > 2:
+        sys.exit(f'EXITING: Too many files given as input {" ".join(inputs)}')
+    elif len(inputs) == 2:
+        input1, input2 = inputs
+        if interleaved:
+            interleaved = False
+        logger.info(f'Extracting barocodes from fastq {input1} and {input2}')
+    elif len(inputs) == 1 and interleaved:
+        input1 = inputs[0]
+        logger.info(f'Extracting barocodes from interleaved fastq {input1}')
+    else:
+        sys.exit(f'EXITING: Files given as input ({inputs}) does not match the required 1 or 2.')
+
+    return input1, input2, interleaved
+
+
+def output_files(output1, output2):
+    """
+    Checks output arguments given and if file should be written as interleaved or not.
+    :param output1: string. Output1 argument
+    :param output2: string. Output2 argument
+    :return: output1, output2, interleaved.
+    """
+    interleaved = bool()
+    if output1 and output2:
+        interleaved = False
+        logger.info(f'Writing paired fastq to files {output1} and {output2}')
+    elif not output1 and not output2:
+        output1 = sys.stdout.buffer
+        interleaved = True
+        logger.info(f'Writing interleaved fastq to stdout.')
+    elif output1 and not output2:
+        interleaved = True
+        logger.info(f'Writing interleaved fastq to file {output1}.')
```

Comment:
I believe this can be  simplified by writing
```suggestion
if output2:
    interleaved = False
    logger.info(f'Writing paired fastq to files {output1} and {output2}')
else:
    interleaved = True 
    if output1:
        output1 = sys.stdout.buffer
        logger.info(f'Writing interleaved fastq to stdout.')
    else:
        logger.info(f'Writing interleaved fastq to file {output1}.')
```

Reply:
Hm, can't seem to get the multiline suggestion feature to work, either way this would replace lines 48-57.